core/src/test/scala/org/genivi/sota/core/CampaignResourceSpec.scala

```diff
@@ -323,4 +355,74 @@ class CampaignResourceSpec extends FunSuite
 
     getCampaignsOk().filter(_.id == id).map(_.createdAt) shouldEqual createdAt
   }
+
+  test("campaign with static delta changes to 'in preparation' after launch") {
+    val campName = CreateCampaignGen.sample.get
+    val id = createCampaignOk(campName)
+
+    val pkgId = createRandomPackage()
+    setPackage(id, pkgId, StatusCodes.OK)
+
+    val setgroups = SetCampaignGroupsGen.sample.get
+    setGroups(id, setgroups, StatusCodes.OK)
+
+    setDeltaFrom(id, Some(createRandomPackage()), StatusCodes.OK)
+
+    val lc = LaunchCampaignGen.sample.get
+
+    launch(id, lc, StatusCodes.NoContent)
+
+    val camp = fetchCampaignOk(id)
+
+    camp.meta.status shouldBe CampaignStatus.InPreparation
+  }
+
+  test("campaign with static delta cannot be edited after launch") {
+    val campName = CreateCampaignGen.sample.get
+    val id = createCampaignOk(campName)
+
+    val pkgId = createRandomPackage()
+    setPackage(id, pkgId, StatusCodes.OK)
+
+    val setgroups = SetCampaignGroupsGen.sample.get
+    setGroups(id, setgroups, StatusCodes.OK)
+
+    setDeltaFrom(id, Some(createRandomPackage()), StatusCodes.OK)
+
+    val lc = LaunchCampaignGen.sample.get
+
+    launch(id, lc, StatusCodes.NoContent)
+
+    setGroups(id, SetCampaignGroupsGen.sample.get, StatusCodes.Locked)
```

Comment:
Ah okay, but it would be nice to make sure that we can't change the static Δ after a launch, so maybe update the test for ` test("can't update launched campaign")` to include a
```
setDeltaFrom(id, Some(createRandomPackage()), StatusCodes.Locked)
```